File: app/controllers/article.go
package controllers

import (
	"github.com/revel/revel"
)

type ArticleController struct {
	ApplicationController
}

type Article struct {
	Slug           string   `json:"slug"`
	Title          string   `json:"title"`
	Description    string   `json:"description"`
	Body           string   `json:"body"`
	Favorited      bool     `json:"favorited"`
	FavoritesCount int      `json:"favoritesCount"`
	TagList        []string `json:"tagList"`
	CreatedAt      string   `json:"createdAt"`
	UpdatedAt      string   `json:"updatedAt"`
	Author         Author   `json:"user"`
}

type Author struct {
	Username  string `json:"username"`
	Bio       string `json:"bio"`
	Image     string `json:"image"`
	Following bool   `json:"following"`
}

type ArticleJSON struct {
	Article `json:"article"`
}

type ArticlesJSON struct {
	Articles      []Article `json:"articles"`
	ArticlesCount int       `json:"articlesCount"`
}

func (c ArticleController) Index() revel.Result {
	return c.Todo()
}
func (c ArticleController) Create() revel.Result {
	return c.Todo()
}
func (c ArticleController) Read() revel.Result {
	return c.Todo()
}
func (c ArticleController) Update() revel.Result {
	return c.Todo()
}
func (c ArticleController) Delete() revel.Result {
	return c.Todo()
}

File: lib/src/articles/model/article.dart
import 'package:dart_shelf_realworld_example_app/src/common/model/base_entity.dart';

class Article extends BaseEntity {
  final String authorId;
  final String slug;
  final String title;
  final String description;
  final String body;
  final List<String> tagList;

  Article(
      {required String id,
      required this.authorId,
      required this.slug,
      required this.title,
      required this.description,
      required this.body,
      required this.tagList,
      required DateTime createdAt,
      required DateTime updatedAt,
      required DateTime? deletedAt})
      : super(
            id: id,
            createdAt: createdAt,
            updatedAt: updatedAt,
            deletedAt: deletedAt);
}
